modules/dnn/src/cuda/functors.hpp

```diff
@@ -44,11 +46,30 @@ struct mish_functor {
     }
 };
 
+template <>
+struct mish_functor<float> {
+    __device__ float operator()(float value) {
+        // f(x) = x * tanh(log1pexp(x));
+        using csl::device::fast_divide;
+        using csl::device::fast_exp;
+
+        auto e = fast_exp(value);
+        if (value <= -18.0f)
+            return value * e;
+
+        auto n = e * e + 2 * e;
```

Comment:
n = e(e+2)

Reply:
`e` is `exp(value)`. For negative values below some threshold, `e` becomes really small that adding it to `2` will give back `2`. This will cause you to lose significant digits. `e * e` is close to `2 * e` in the small negative numbers range `[-1, -15]` and hence you won't suffer from the same problem.

While this is algebraically same, it's not the same when working with finite-precision numbers.

Reply:
Oh your comment disappeared somehow. I agree that what you say makes sense. So do you think that creating specific taylor series function for some functions like mish would even more boost speed?

Reply:
I deleted it because it was wrong. `e * e + 2 * e` and `e * (e + 2)` are different but the error  was only a few ULPs in the final expression and it happened at the very edge. I have to do more rigirous checks. It's negligible improvement though.

I don't think Taylor series would help. It would still involve `tanh`, `log`, `exp`, etc. It's possible to make a rational approximation to mish but that computationally more expensive than what's currently in the PR. And it improves the accuracy negligibly, so it's not worth it.

Reply:
`e * e + 2 * e` leads to `FMA(e, e, FADD(e, e))`
`e * (e + 2)` leads to `FMUL(e, FADD(e + 2))`